kitsune/wiki/tests/test_templates.py

```diff
@@ -1013,6 +1013,143 @@ def test_keywords_require_permission(self):
         new_rev = Revision.objects.filter(document=doc).order_by('-id')[0]
         eq_('keyword1 keyword2', new_rev.keywords)
 
+    def test_restore_draft_revision(self):
+        rev = ApprovedRevisionFactory(is_ready_for_localization=True)
+        doc = rev.document
+        doc.is_localizable = True
+        doc.save()
+        # Check the Translation page has Save Draft Button
+        trans_url = reverse('wiki.translate', locale='bn-BD', args=[doc.slug])
+        trans_resp = self.client.get(trans_url)
+        eq_(200, trans_resp.status_code)
+        trans_content = pq(trans_resp.content)
+        eq_(0, len(trans_content('.user-messages li')))
+        eq_(1, len(trans_content('.submit .btn-draft')))
+        data = {
+            'content': 'Test Content bla bla bla',
+            'keywords': 'something',
+            'summary': 'some summary',
+            'title': 'test title',
+            'slug': 'test-slug',
+            'based_on': int(rev.id)}
+        # Make a Draft with above mentioned data for bn-BD locale
+        draft_url = reverse('wiki.draft_revision', locale='bn-BD')
+        draft_resp = self.client.post(draft_url, data)
+        eq_(201, draft_resp.status_code)
+        # Get the Translation page
+        cache.clear()
+        trans_resp = self.client.get(trans_url)
+        trans_content = pq(trans_resp.content)
+        # Check user message is shown there
+        eq_(1, len(trans_content('.user-messages li')))
+        # Check there are two buttons for restore and discard
+        eq_(2, len(trans_content('.user-messages .info form .btn')))
+        # Restore with the draft data
+        draft_request = {'restore': 'Restore'}
+        cache.clear()
```

Comment:
I dont know. While requesting the same page in same test, it raised error for me because it was cached. The translation page get cached and when I tried to get the page again, it returned the cached page. 
If test passes without clearing the cache, I am happy to remove it